adapter/src/dummy.rs

```diff
@@ -21,33 +21,40 @@ impl ChannelValidator for DummyAdapter {}
 impl Adapter for DummyAdapter {
     type Output = DummyAdapter;
 
-    fn init(opts: AdapterOptions, _config: &Config) -> DummyAdapter {
-        // opts.dummy_identity.expect("dummyIdentity required");
-        // opts.dummy_auth.expect("dummy auth required");
-        // opts.dummy_auth_tokens.expect("dummy auth tokens required");
-        // self.identity = opts.dummy_identity.unwrap();
-        // self.authTokens = opts.dummy_auth.unwrap();
-        // self.verifiedAuth = opts.dummy_auth_tokens.unwrap();
-        Self {
-            identity: opts.dummy_identity.unwrap(),
-            auth_tokens: HashMap::new(),
-            verified_auth: HashMap::new(),
-        }
+    fn init(opts: AdapterOptions, config: &Config) -> AdapterResult<DummyAdapter> {
+        let (identity, authorization_tokens, session_tokens) =
+            match (opts.dummy_identity, opts.dummy_auth, opts.dummy_auth_tokens) {
+                (Some(identity), Some(authorization_tokens), Some(session_tokens)) => {
+                    (identity, authorization_tokens, session_tokens)
+                }
+                (_, _, _) => {
+                    return Err(AdapterError::Configuration(
+                        "dummy_identity, dummy_auth, dummy_auth_tokens required".to_string(),
+                    ))
+                }
+            };
+
+        Ok(Self {
+            identity,
+            config: config.to_owned(),
+            session_tokens,
+            authorization_tokens,
+        })
     }
 
     fn unlock(&self) -> AdapterResult<bool> {
         Ok(true)
     }
 
-    fn whoami(&self) -> String {
-        self.identity.to_string()
+    fn whoami(&self) -> AdapterResult<String> {
```

Comment:
Can you document somewhere, what is the use case when you don't have an identity for the adapter?

Reply:
I don't think it's possible to not have an identity for the adapter.
The reason behind the AdapterResult is the because of the Ethereum adapter where you can't call `whoami()` without unlocking the adapter first